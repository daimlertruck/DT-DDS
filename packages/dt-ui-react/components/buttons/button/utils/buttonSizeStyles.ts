import { CustomTheme as Theme } from '../../../../types';
import { ButtonSizes } from '../constants/types';

export const buttonSizeStyles = (size: ButtonSizes, theme: Theme) =>
  ({
    large: `
      ${theme.fontStyles.button1}
      padding: ${theme.spacing['2xs']} ${theme.spacing.s};
    `,
    medium: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['3xs']} ${theme.spacing.xs};
    `,
    small: `
      ${theme.fontStyles.button2}
      padding: ${theme.spacing['4xs']} ${theme.spacing['2xs']};
    `,
  }[size]);
