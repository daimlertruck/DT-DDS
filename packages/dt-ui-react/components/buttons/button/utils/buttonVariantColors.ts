import { CustomTheme as Theme } from '../../../../types';
import {
  ButtonVariant,
  ContainedColors,
  OutlinedColors,
  TextColors,
} from '../constants/types';

interface ColorKeys {
  color: string;
  bgColor: string;
  hoverBgColor?: string;
  hoverColor?: string;
}

export const buttonVariantColors = (
  color: ContainedColors | OutlinedColors,
  theme: Theme
): Record<ButtonVariant, ColorKeys> => ({
  contained: {
    color: theme.colors.white,
    bgColor: theme.palette[color],
    hoverBgColor: theme.palette[`${color}Dark_300` as ContainedColors],
  },
  outlined: {
    color: color === 'neutral' ? theme.colors.grey_200 : theme.palette[color],
    bgColor: theme.colors.white,
    hoverBgColor: theme.palette[`${color}Light_300` as OutlinedColors],
  },
  text: {
    color: theme.palette[color],
    bgColor: 'transparent',
    hoverColor: theme.palette[`${color as TextColors}Dark_300`],
  },
});
