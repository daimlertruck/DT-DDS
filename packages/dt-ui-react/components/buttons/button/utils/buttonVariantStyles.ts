import { CustomTheme as Theme } from '../../../../types';
import { ButtonVariant, OutlinedColors, SolidColors } from '../constants/types';

export const buttonVariantStyles = (
  variant: ButtonVariant,
  theme: Theme,
  color: SolidColors | OutlinedColors
) =>
  ({
    solid: `
      text-transform: uppercase;
      border: ${theme.spacing.none};

      &:focus {
        box-shadow: 0px 2px 1px 0px #1A1A1A0D, 0px 0px 0px 4px ${theme.colors.blue_10};
        background-color: ${theme.palette[color]};
      }

      &:disabled {
        opacity: 38%;
      }
  `,
    outlined: `
      border: 1px solid;
      text-transform: uppercase;
        ${
          color === 'neutral'
            ? theme.palette.neutralDark_100
            : theme.palette[color]
        };

      &:disabled {
        color: ${
          color === 'neutral'
            ? theme.palette.neutralLight_100
            : theme.palette[color]
        };
        background-color: ${theme.colors.white};
        border-color: ${
          color === 'neutral'
            ? theme.palette.neutralLight_100
            : theme.palette[color]
        };
        cursor: default;
      }
    `,
    text: `
      border: none;

      &:disabled {
        color: ${theme.palette.neutralLight_100};
      }
    `,
  }[variant]);
