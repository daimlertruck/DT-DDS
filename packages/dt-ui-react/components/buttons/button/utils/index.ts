export * from './buttonVariantColors';
export * from './buttonVariantStyles';
