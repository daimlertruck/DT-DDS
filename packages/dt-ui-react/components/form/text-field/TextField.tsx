import { BaseProps } from '@dt-ui/react-core';
import { LabelField } from '@dt-ui/react-label-field';
import { Spinner } from '@dt-ui/react-spinner';
import { Typography } from '@dt-ui/react-typography';
import React, {
  useState,
  useEffect,
  ChangeEvent,
  MutableRefObject,
  HTMLInputTypeAttribute,
  ReactNode,
} from 'react';

import { ErrorIcon } from '../../../core/assets';

import {
  TextFieldStyled,
  InputFieldStyled,
  InputFieldIconStyled,
  TextFieldMessageStyled,
} from './TextField.styled';

export interface TextFieldProps extends BaseProps {
  label: string;
  isDisabled?: boolean;
  isLoading?: boolean;
  type?: HTMLInputTypeAttribute;
  hasError?: boolean;
  name?: string;
  required?: boolean;
  initialValue?: string;
  maxLength?: number;
  inputRef?: MutableRefObject<HTMLInputElement>;
  message?: string | null;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  icon?: ReactNode;
}

export const TextField = ({
  hasError: hasErrorProp = false,
  isDisabled,
  isLoading,
  label,
  name,
  required,
  style,
  children,
  initialValue,
  inputRef,
  maxLength,
  message: messageProp = '',
  type = 'text',
  icon,
  onChange = () => null,
}: TextFieldProps) => {
  const [activeInput, setActiveInput] = useState(false);
  const [inputValue, setInputValue] = useState('');
  const [hasError, setHasError] = useState(hasErrorProp);
  const [message, setMessage] = useState(messageProp);
  const id = label.replaceAll(' ', '-');

  useEffect(() => {
    // Check if there's an initial value coming from props
    if (!!initialValue) {
      setActiveInput(true);
      setInputValue(initialValue);
    } else {
      setActiveInput(false);
      setInputValue('');
    }
  }, [initialValue]);

  useEffect(() => {
    setHasError(hasErrorProp);
    setMessage(messageProp);

    if (!inputValue) {
      setHasError(false);
    }
  }, [hasErrorProp, initialValue, messageProp, inputValue]);

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue(event.target.value);
    if (onChange) {
      onChange(event);
    }
  };

  const onFocus = () => {
    setActiveInput(true);
  };

  const onBlur = (event: ChangeEvent<HTMLInputElement>) => {
    if (event.currentTarget.value === '') {
      setActiveInput(false);

      if (required) {
        setHasError(true);
        setMessage('This field is required.');
      }
    }

    if (required && !!event.currentTarget.value && !hasError) {
      setHasError(false);
    }

    inputValue.length > 0 ? setActiveInput(true) : setActiveInput(false);
  };

  return (
    <TextFieldStyled style={style}>
      <LabelField forId={id} isActive={activeInput} isDisabled={isDisabled}>
        {label} {required ? '*' : null}
      </LabelField>

      {hasError || isLoading || !!icon ? (
        <InputFieldIconStyled>
          {hasError ? <ErrorIcon /> : null}
          {isLoading ? <Spinner colorScheme='negative' size='small' /> : null}
          {!isLoading && !!icon && icon}
        </InputFieldIconStyled>
      ) : null}
      <InputFieldStyled
        disabled={isDisabled}
        hasError={hasError}
        id={id}
        maxLength={maxLength}
        name={name || id}
        onBlur={onBlur}
        onChange={handleChange}
        onFocus={onFocus}
        ref={inputRef}
        required={required}
        type={type}
        value={inputValue}
      />
      {message ? (
        <TextFieldMessageStyled>
          <Typography
            color={hasError ? 'error.default' : 'content.body'}
            element='span'
            fontStyles='pXXSmall'
          >
            {message}
          </Typography>
        </TextFieldMessageStyled>
      ) : null}
      {children}
    </TextFieldStyled>
  );
};
