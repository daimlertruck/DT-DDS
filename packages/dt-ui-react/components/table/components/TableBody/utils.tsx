import { Children, ReactNode, cloneElement, isValidElement } from 'react';

import { TableDataCellHeader } from '../TableDataCell/TableDataCell';

export const addHeadersToDataCells = (
  dataCells: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(dataCells, (dataCell, index) => {
    if (isValidElement(dataCell)) {
      const header = headers[index];
      return cloneElement(
        dataCell,
        {},
        <>
          <TableDataCellHeader>{header}</TableDataCellHeader>
          {dataCell.props.children}
        </>
      );
    }
    return dataCell;
  });
};

export const addHeadersToDataCellsInRows = (
  rows: ReactNode,
  headers: ReactNode[]
) => {
  return Children.map(rows, (row) => {
    if (isValidElement(row)) {
      return cloneElement(
        row,
        {},
        addHeadersToDataCells(row.props.children, headers)
      );
    }
    return row;
  });
};
