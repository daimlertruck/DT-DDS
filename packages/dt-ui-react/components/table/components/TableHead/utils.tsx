import { Children, ReactNode, isValidElement } from 'react';

const extractColumnHeaders = (columnHeaders: ReactNode): ReactNode[] => {
  return Children.toArray(columnHeaders).reduce<ReactNode[]>(
    (headers, columnHeader) => {
      if (isValidElement(columnHeader)) {
        return [...headers, columnHeader.props.children];
      }

      return headers;
    },
    []
  );
};

export const extractTableHeaders = (theadRow: ReactNode) => {
  if (isValidElement(theadRow)) {
    return extractColumnHeaders(theadRow.props.children);
  }

  return [];
};
