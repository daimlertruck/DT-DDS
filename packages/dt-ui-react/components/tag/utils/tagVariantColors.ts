import { CustomTheme as Theme } from '../../../types';
import { TagColor, TagVariant } from '../constants';

interface TagVariantColors {
  border: string | number;
  color: string;
  borderColor?: string;
  backgroundColor: string;
  '&:hover'?: {
    borderColor?: string;
    backgroundColor: string;
    color: string;
  };
}

interface TagVariantColorsProps {
  theme: Theme;
  color: TagColor;
  variant: TagVariant;
  hasHover: boolean;
  isDisabled: boolean;
}

const getColorWithShade = ({
  theme,
  color,
  shade,
}: Pick<TagVariantColorsProps, 'color'> & {
  shade: '90' | '60' | '30' | '10';
  theme: Theme;
}) => {
  const { colors, palette } = theme;

  if (color === 'primary') {
    return {
      '90': palette.primaryDark_500,
      '60': palette.primaryDark_200,
      '30': palette.primaryLight_100,
      '10': palette.primaryLight_400,
    }[shade];
  }

  const colorWithShade: keyof Theme['colors'] = `${color}_${shade}`;

  return colors[colorWithShade];
};

export const tagVariantColors = ({
  theme,
  color,
  variant,
  hasHover,
  isDisabled,
}: TagVariantColorsProps): TagVariantColors => {
  const colorShade90 = getColorWithShade({
    theme,
    color,
    shade: '90',
  });
  const colorShade60 = getColorWithShade({
    theme,
    color,
    shade: '60',
  });
  const colorShade30 = getColorWithShade({
    theme,
    color,
    shade: '30',
  });
  const colorShade10 = getColorWithShade({
    theme,
    color,
    shade: '10',
  });

  const colorWithDisabled = isDisabled ? colorShade30 : colorShade60;

  return {
    solid: {
      border: theme.spacing.none,
      color: theme.colors.white,
      backgroundColor: colorWithDisabled,
      ...(hasHover && {
        '&:hover': {
          backgroundColor: colorShade90,
          color: theme.colors.white,
        },
      }),
    },
    outlined: {
      border: `1px solid ${colorShade60}`,
      color: colorWithDisabled,
      backgroundColor: colorShade10,
      borderColor: colorWithDisabled,
      ...(hasHover && {
        '&:hover': {
          borderColor: colorShade90,
          backgroundColor: colorShade10,
          color: colorShade90,
        },
      }),
    },
  }[variant];
};
