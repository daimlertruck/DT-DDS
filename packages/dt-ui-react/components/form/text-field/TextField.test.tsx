import { act, render, screen, fireEvent } from '@testing-library/react';
import React from 'react';

import { withProviders } from '../../../utils';

import { TextField } from './TextField';

describe('<TextField /> component', () => {
  const ProvidedTextField = withProviders(TextField);

  const hasInputValue = (e: HTMLElement, inputValue: string) => {
    return screen.getByDisplayValue(inputValue) === e;
  };

  const renderInput = () => {
    const { container } = render(
      <ProvidedTextField label='My Awesome Input' />
    );
    const input = container.querySelector('input') as HTMLElement;
    return { input };
  };

  it('renders input with an empty initial value', () => {
    const { input } = renderInput();
    expect(input).toHaveValue('');
  });

  it('renders input with an initial value', () => {
    render(
      <ProvidedTextField
        initialValue='Initial Value'
        label='My Input with prefilled value'
      />
    );

    const input = screen.getByLabelText('My Input with prefilled value');
    expect(hasInputValue(input, 'Initial Value')).toBe(true);
  });

  it('fills input correctly with new value on change Event', () => {
    const { input } = renderInput();
    fireEvent.change(input, { target: { value: 'New input value' } });

    expect(hasInputValue(input, 'New input value')).toBe(true);
  });

  it('should render the TextField with error styles and message', () => {
    const { container } = render(
      <ProvidedTextField
        hasError
        initialValue='Initial Value'
        label='My Input with Error Message'
        message='My Error Message'
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('should render the TextField with info message', () => {
    const { container } = render(
      <ProvidedTextField
        hasError={false}
        label='My Input with info message'
        message='My Infor Message'
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('renders disabled input', () => {
    const { container } = render(
      <ProvidedTextField isDisabled label='My disabled input' />
    );

    expect(container).toMatchSnapshot();
  });

  it('renders hiddent text input', () => {
    const { container } = render(
      <ProvidedTextField label='Hidden text' type='password' />
    );

    expect(container.querySelector('input')).toHaveAttribute(
      'type',
      'password'
    );
  });

  it('renders required text field', () => {
    const { container } = render(
      <ProvidedTextField label='Required text' required />
    );

    expect(container.querySelector('label')).toHaveTextContent(
      'Required text *'
    );
  });

  it('should have active state on focus', () => {
    const { container } = render(<ProvidedTextField label='Some text' />);

    const input = container.querySelector('input') as HTMLElement;
    const label = container.querySelector('label') as HTMLElement;

    act(() => {
      input.focus();
    });

    expect(label).toHaveStyle('font-size: .75rem');
    expect(label).toHaveStyle('transform: translateY(-45%)');
  });

  it('renders Loading input', () => {
    const { container } = render(
      <ProvidedTextField isLoading label='My disabled input' />
    );

    expect(container).toMatchSnapshot();
  });

  describe('onBlur event', () => {
    it('should have error state with error message', () => {
      const { container } = render(
        <ProvidedTextField label='Some text' required />
      );
      const input = container.querySelector('input') as HTMLElement;

      fireEvent.blur(input, { currentTarget: { value: '' } });

      expect(input).toHaveStyle('border-bottom: 2px solid #FF4A4A');
      expect(container).toMatchSnapshot();
    });

    it('should have active state', () => {
      const { container } = render(
        <ProvidedTextField initialValue='Value' label='Some text' required />
      );
      const input = container.querySelector('input') as HTMLElement;
      const label = container.querySelector('label') as HTMLElement;

      fireEvent.blur(input, { currentTarget: { value: 'Some value' } });

      expect(label).toHaveStyle('font-size: .75rem');
      expect(label).toHaveStyle('transform: translateY(-45%)');
    });
  });

  it('renders input with side button', () => {
    const button = <button type='button'>My button</button>;
    const { container } = render(
      <ProvidedTextField icon={button} label='My input' />
    );

    expect(container).toMatchSnapshot();
  });
});
