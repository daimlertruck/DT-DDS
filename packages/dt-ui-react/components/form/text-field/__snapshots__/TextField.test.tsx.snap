// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> component onBlur event should have error state with error message 1`] = `
.emotion-0 {
  display: block;
  height: auto;
  width: 100%;
  position: relative;
}

.emotion-2 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  color: #8B8C8C;
  cursor: inherit;
  font-size: 1rem;
  left: 15px;
  top: 15px;
  position: absolute;
  -webkit-transform: translateY(0);
  -moz-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  -webkit-transition: -webkit-transform 150ms ease-out,font-size 150ms ease-out;
  transition: transform 150ms ease-out,font-size 150ms ease-out;
}

.emotion-4 {
  position: absolute;
  right: 15px;
  top: 15px;
  width: 24px;
  height: 24px;
  color: #D21C1C;
}

.emotion-6 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  background-color: #FFFFFF;
  border: 1px solid #323232;
  border-bottom: 2px solid #D21C1C;
  color: #151617;
  min-width: 198px;
  padding: 20px 15px 5px 15px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  height: 53px;
}

.emotion-8 {
  padding-left: 15px;
}

.emotion-10 {
  font-weight: 400;
  font-size: 12px;
  line-height: 16.08px;
  letter-spacing: normal;
  margin: 0;
  color: #D21C1C;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <label
      class="emotion-2 emotion-3"
      data-testid="label-field"
      for="Some-text"
    >
      Some text
       
      *
    </label>
    <span
      class="emotion-4 emotion-5"
    >
      <svg
        color="#D21C1C"
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.0711 19.0711C17.1957 20.9464 14.6522 22 12 22C9.34784 22 6.8043 20.9464 4.92893 19.0711C3.05357 17.1957 2 14.6522 2 12C2 9.34784 3.05357 6.8043 4.92893 4.92893C6.8043 3.05357 9.34784 2 12 2C14.6522 2 17.1957 3.05357 19.0711 4.92893C20.9464 6.8043 22 9.34784 22 12C22 14.6522 20.9464 17.1957 19.0711 19.0711Z"
          fill="currentColor"
        />
        <path
          d="M12.8797 13.6863L13.1157 6.48828H10.8997L11.0997 13.6863H12.8797ZM11.9907 17.5123C12.1621 17.5181 12.333 17.4894 12.4932 17.4281C12.6534 17.3667 12.7997 17.2739 12.9234 17.1551C13.0471 17.0362 13.1458 16.8938 13.2136 16.7362C13.2813 16.5786 13.3168 16.4091 13.318 16.2375C13.3191 16.066 13.2859 15.8959 13.2202 15.7375C13.1545 15.579 13.0577 15.4353 12.9356 15.3148C12.8135 15.1944 12.6684 15.0996 12.5091 15.0361C12.3497 14.9727 12.1792 14.9418 12.0077 14.9453C11.6733 14.9521 11.3547 15.0892 11.1198 15.3274C10.8849 15.5656 10.7522 15.886 10.75 16.2205C10.7478 16.555 10.8762 16.8772 11.1079 17.1184C11.3397 17.3597 11.6564 17.501 11.9907 17.5123Z"
          fill="white"
        />
      </svg>
    </span>
    <input
      class="emotion-6 emotion-7"
      id="Some-text"
      name="Some-text"
      required=""
      type="text"
      value=""
    />
    <div
      class="emotion-8 emotion-9"
    >
      <span
        class="emotion-10 emotion-11"
        color="red_100"
        data-testid="typography"
      >
        This field is required.
      </span>
    </div>
  </div>
</div>
`;

exports[`<TextField /> component renders Loading input 1`] = `
@keyframes animation-0 {
  from {
    -webkit-transform: rotate(0);
    -moz-transform: rotate(0);
    -ms-transform: rotate(0);
    transform: rotate(0);
  }

  to {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-0 {
  display: block;
  height: auto;
  width: 100%;
  position: relative;
}

.emotion-2 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  color: #8B8C8C;
  cursor: inherit;
  font-size: 1rem;
  left: 15px;
  top: 15px;
  position: absolute;
  -webkit-transform: translateY(0);
  -moz-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  -webkit-transition: -webkit-transform 150ms ease-out,font-size 150ms ease-out;
  transition: transform 150ms ease-out,font-size 150ms ease-out;
}

.emotion-4 {
  position: absolute;
  right: 15px;
  top: 15px;
  width: 24px;
  height: 24px;
  color: #D21C1C;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-8 {
  width: 32px;
  height: 32px;
  border: 2px solid #FFFFFF;
  border-radius: 50%;
  border-top: 2px solid #002441;
  -webkit-animation: animation-0 1s cubic-bezier(0.3, 0.1, 0.25, 0.65) 0s infinite normal none;
  animation: animation-0 1s cubic-bezier(0.3, 0.1, 0.25, 0.65) 0s infinite normal none;
}

.emotion-10 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  background-color: #FFFFFF;
  border: 1px solid #323232;
  border-bottom: 1px solid #323232;
  color: #151617;
  min-width: 198px;
  padding: 20px 15px 5px 15px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  height: 53px;
}

.emotion-10:focus {
  border-bottom: 2px solid #002441;
  outline: none;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <label
      class="emotion-2 emotion-3"
      data-testid="label-field"
      for="My-disabled-input"
    >
      My disabled input
       
    </label>
    <span
      class="emotion-4 emotion-5"
    >
      <div
        class="emotion-6 emotion-7"
        style="margin: auto;"
      >
        <div
          class="emotion-8 emotion-9"
        />
      </div>
    </span>
    <input
      class="emotion-10 emotion-11"
      id="My-disabled-input"
      name="My-disabled-input"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`<TextField /> component renders disabled input 1`] = `
.emotion-0 {
  display: block;
  height: auto;
  width: 100%;
  position: relative;
}

.emotion-2 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  color: #323232;
  cursor: inherit;
  font-size: 1rem;
  left: 15px;
  top: 15px;
  position: absolute;
  -webkit-transform: translateY(0);
  -moz-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  -webkit-transition: -webkit-transform 150ms ease-out,font-size 150ms ease-out;
  transition: transform 150ms ease-out,font-size 150ms ease-out;
}

.emotion-4 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  background-color: #636464;
  border: 1px solid #323232;
  border-bottom: 1px solid #323232;
  color: #8B8C8C;
  min-width: 198px;
  padding: 20px 15px 5px 15px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  height: 53px;
}

.emotion-4:focus {
  border-bottom: 2px solid #002441;
  outline: none;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <label
      class="emotion-2 emotion-3"
      data-testid="label-field"
      disabled=""
      for="My-disabled-input"
    >
      My disabled input
       
    </label>
    <input
      class="emotion-4 emotion-5"
      disabled=""
      id="My-disabled-input"
      name="My-disabled-input"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`<TextField /> component renders input with side button 1`] = `
.emotion-0 {
  display: block;
  height: auto;
  width: 100%;
  position: relative;
}

.emotion-2 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  color: #8B8C8C;
  cursor: inherit;
  font-size: 1rem;
  left: 15px;
  top: 15px;
  position: absolute;
  -webkit-transform: translateY(0);
  -moz-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  -webkit-transition: -webkit-transform 150ms ease-out,font-size 150ms ease-out;
  transition: transform 150ms ease-out,font-size 150ms ease-out;
}

.emotion-4 {
  position: absolute;
  right: 15px;
  top: 15px;
  width: 24px;
  height: 24px;
  color: #D21C1C;
}

.emotion-6 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  background-color: #FFFFFF;
  border: 1px solid #323232;
  border-bottom: 1px solid #323232;
  color: #151617;
  min-width: 198px;
  padding: 20px 15px 5px 15px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  height: 53px;
}

.emotion-6:focus {
  border-bottom: 2px solid #002441;
  outline: none;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <label
      class="emotion-2 emotion-3"
      data-testid="label-field"
      for="My-input"
    >
      My input
       
    </label>
    <span
      class="emotion-4 emotion-5"
    >
      <button>
        My button
      </button>
    </span>
    <input
      class="emotion-6 emotion-7"
      id="My-input"
      name="My-input"
      type="text"
      value=""
    />
  </div>
</div>
`;

exports[`<TextField /> component should render the TextField with error styles and message 1`] = `
.emotion-0 {
  display: block;
  height: auto;
  width: 100%;
  position: relative;
}

.emotion-2 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  color: #8B8C8C;
  cursor: inherit;
  font-size: .75rem;
  left: 15px;
  top: 15px;
  position: absolute;
  -webkit-transform: translateY(-45%);
  -moz-transform: translateY(-45%);
  -ms-transform: translateY(-45%);
  transform: translateY(-45%);
  -webkit-transition: -webkit-transform 150ms ease-out,font-size 150ms ease-out;
  transition: transform 150ms ease-out,font-size 150ms ease-out;
}

.emotion-4 {
  position: absolute;
  right: 15px;
  top: 15px;
  width: 24px;
  height: 24px;
  color: #D21C1C;
}

.emotion-6 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  background-color: #FFFFFF;
  border: 1px solid #323232;
  border-bottom: 2px solid #D21C1C;
  color: #151617;
  min-width: 198px;
  padding: 20px 15px 5px 15px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  height: 53px;
}

.emotion-8 {
  padding-left: 15px;
}

.emotion-10 {
  font-weight: 400;
  font-size: 12px;
  line-height: 16.08px;
  letter-spacing: normal;
  margin: 0;
  color: #D21C1C;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <label
      class="emotion-2 emotion-3"
      data-testid="label-field"
      for="My-Input-with-Error-Message"
    >
      My Input with Error Message
       
    </label>
    <span
      class="emotion-4 emotion-5"
    >
      <svg
        color="#D21C1C"
        fill="none"
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.0711 19.0711C17.1957 20.9464 14.6522 22 12 22C9.34784 22 6.8043 20.9464 4.92893 19.0711C3.05357 17.1957 2 14.6522 2 12C2 9.34784 3.05357 6.8043 4.92893 4.92893C6.8043 3.05357 9.34784 2 12 2C14.6522 2 17.1957 3.05357 19.0711 4.92893C20.9464 6.8043 22 9.34784 22 12C22 14.6522 20.9464 17.1957 19.0711 19.0711Z"
          fill="currentColor"
        />
        <path
          d="M12.8797 13.6863L13.1157 6.48828H10.8997L11.0997 13.6863H12.8797ZM11.9907 17.5123C12.1621 17.5181 12.333 17.4894 12.4932 17.4281C12.6534 17.3667 12.7997 17.2739 12.9234 17.1551C13.0471 17.0362 13.1458 16.8938 13.2136 16.7362C13.2813 16.5786 13.3168 16.4091 13.318 16.2375C13.3191 16.066 13.2859 15.8959 13.2202 15.7375C13.1545 15.579 13.0577 15.4353 12.9356 15.3148C12.8135 15.1944 12.6684 15.0996 12.5091 15.0361C12.3497 14.9727 12.1792 14.9418 12.0077 14.9453C11.6733 14.9521 11.3547 15.0892 11.1198 15.3274C10.8849 15.5656 10.7522 15.886 10.75 16.2205C10.7478 16.555 10.8762 16.8772 11.1079 17.1184C11.3397 17.3597 11.6564 17.501 11.9907 17.5123Z"
          fill="white"
        />
      </svg>
    </span>
    <input
      class="emotion-6 emotion-7"
      id="My-Input-with-Error-Message"
      name="My-Input-with-Error-Message"
      type="text"
      value="Initial Value"
    />
    <div
      class="emotion-8 emotion-9"
    >
      <span
        class="emotion-10 emotion-11"
        color="red_100"
        data-testid="typography"
      >
        My Error Message
      </span>
    </div>
  </div>
</div>
`;

exports[`<TextField /> component should render the TextField with info message 1`] = `
.emotion-0 {
  display: block;
  height: auto;
  width: 100%;
  position: relative;
}

.emotion-2 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  color: #8B8C8C;
  cursor: inherit;
  font-size: 1rem;
  left: 15px;
  top: 15px;
  position: absolute;
  -webkit-transform: translateY(0);
  -moz-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  -webkit-transition: -webkit-transform 150ms ease-out,font-size 150ms ease-out;
  transition: transform 150ms ease-out,font-size 150ms ease-out;
}

.emotion-4 {
  font-weight: 400;
  font-size: 16px;
  line-height: 21.44px;
  letter-spacing: normal;
  margin: 0;
  background-color: #FFFFFF;
  border: 1px solid #323232;
  border-bottom: 1px solid #323232;
  color: #151617;
  min-width: 198px;
  padding: 20px 15px 5px 15px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  width: 100%;
  height: 53px;
}

.emotion-4:focus {
  border-bottom: 2px solid #002441;
  outline: none;
}

.emotion-6 {
  padding-left: 15px;
}

.emotion-8 {
  font-weight: 400;
  font-size: 12px;
  line-height: 16.08px;
  letter-spacing: normal;
  margin: 0;
  color: #8B8C8C;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <label
      class="emotion-2 emotion-3"
      data-testid="label-field"
      for="My-Input-with-info-message"
    >
      My Input with info message
       
    </label>
    <input
      class="emotion-4 emotion-5"
      id="My-Input-with-info-message"
      name="My-Input-with-info-message"
      type="text"
      value=""
    />
    <div
      class="emotion-6 emotion-7"
    >
      <span
        class="emotion-8 emotion-9"
        color="grey_100"
        data-testid="typography"
      >
        My Infor Message
      </span>
    </div>
  </div>
</div>
`;
